### PR TITLE
[DB/Quest] Fixed To Greymane Manor - Quest: 14465 (#28)

### LINKED ISSUE #28: To Greymane Manor - Quest: 14465
### Current Behaviour

Neither the npc nor the horses appear from the beginning to the end of the mission it should be noted that you have to put gm on to visualize the npc and complete the mission

### Expected Blizzlike Behaviour

When arriving at the mansion, the npc should be visible, in order to continue with the chain of missions.

### Source

- https://www.wowhead.com/quest=14465/to-greymane-manor
- https://www.wowhead.com/npc=36452/gwen-armstead
- https://www.wowhead.com/npc=36606/queen-mia-greymane
- https://www.wowhead.com/spell=82892/update-bind-to-greymane-manor
- https://www.wowhead.com/spell=69254/carriage-ride
- https://www.wowhead.com/spell=69255/summon-swift-mountain-horse
- https://www.wowhead.com/spell=69256/forcecast-summon-swift-mountain-horse
- https://www.wowhead.com/npc=36741/swift-mountain-horse

### Steps to reproduce the problem

1. Take the quest
2. Go to the castle to complete it
(Although the castle should be reached by horse).
- https://medal.tv/es/games/requested/clips/y4rmA-_UJTeGb/d1337Ade0NmB

### Extra Notes

None

### FC rev. hash/commit

https://github.com/FirelandsProject/firelands-cata/commit/dce019da37cbddd4752723d17d909832fa34ee05

### Operating system

Windows

### Custom changes or Modules

None

### PR DESCRIPTION
**Pull request by kuramajssken001**
**Changes proposed**
Fix To Greymane Manor [Quest: 14465]

**Tests performed**
In-Game Test

**How to Test the Changes:**

Download the branch.
Build
Create a wolf
Start Quests until 14465 quest.
try to complete

### CODE REVIEW

**Comment by pangolp** [comment]
Let me know when it is ready to be reviewed.

File: data/sql/updates/pending_db_world/rev_1662378898044710600.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
+(36452, 0, 0, 1, 19, 0, 100, 0, 14465, 0, 0, 0, 0, 85, 69256, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Gwen Armstead SAI Script'),
+(36452, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 82892, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Gwen Armstead SAI Script'),
+(36452, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 85, 105011, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Gwen Armstead SAI Script'),
+(36741, 0, 0, 0, 60, 0, 100, 513, 1000, 1000, 0, 0, 0, 53, 1, 36741, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Swift Mountain Horse SAI Script'),
+(36741, 0, 1, 0, 40, 0, 100, 512, 2, 36741, 0, 0, 0, 9, 0, 0, 0, 0, 0, 0, 14, 9956226, 196399, 0, 0, 0, 0, 0, 'Swift Mountain Horse SAI Script'),
```
Comment (SargeroDeV) [inline data/sql/updates/pending_db_world/rev_1662378898044710600.sql:8]: EN:
event 40 has changed to the way it was before now event_param2 only uses a pathID if it is a pointid of a waypoint_data
ES:
ya el evento 40 cambio a como era anteriormente ahora el event_param2 solo usa un pathID si es un pointid de un waypoint_data

File: data/sql/updates/pending_db_world/rev_1662378898044710600.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+(36741, 0, 2, 0, 40, 0, 100, 512, 3, 36741, 0, 0, 0, 9, 0, 0, 0, 0, 0, 0, 14, 9956226, 196399, 0, 0, 0, 0, 0, 'Swift Mountain Horse SAI Script'),
```
Comment (SargeroDeV) [inline data/sql/updates/pending_db_world/rev_1662378898044710600.sql:9]: EN:
the guides of lines 1 and 2 from sai to npc 36741 are wrong, they do not exist in this db.
ES:
las guids de las lineas 1 y 2 del sai al npc 36741 estan erroneas no existen en esta db.

**Comment by SargeroDeV** [comment]
EN: For the quest to work you need to add the phases to be able to see the npc to whom you deliver the reward mission.
ES: Para que la quest funcione necesitas a;adirle las phases para poder ver al npc al que le entregas la mision completada.
```sql
DELETE FROM `phase_area` WHERE `AreaId`=4714 AND `PhaseId`=184;
DELETE FROM `phase_area` WHERE `AreaId`=4814 AND `PhaseId`=184;
INSERT INTO `phase_area` (`AreaId`, `PhaseId`, `Comment`) VALUES (4714, 184, 'need conditions');
INSERT INTO `phase_area` (`AreaId`, `PhaseId`, `Comment`) VALUES (4814, 184, 'need conditions');
```

**Comment by pangolp** [comment]
> EN: For the quest to work you need to add the phases to be able to see the npc to whom you deliver the reward mission. ES: Para que la quest funcione necesitas a;adirle las phases para poder ver al npc al que le entregas la mision completada.
> 
> ```sql
> DELETE FROM `phase_area` WHERE `AreaId`=4714 AND `PhaseId`=184;
> DELETE FROM `phase_area` WHERE `AreaId`=4814 AND `PhaseId`=184;
> INSERT INTO `phase_area` (`AreaId`, `PhaseId`, `Comment`) VALUES (4714, 184, 'need conditions');
> INSERT INTO `phase_area` (`AreaId`, `PhaseId`, `Comment`) VALUES (4814, 184, 'need conditions');
> ```

Anyway, the idea is that it does not make several DELETE or several INSERT either, beyond the things that you mentioned. The idea is that it groups them, whenever possible, to reduce the amount of queries, inside the emulator.

**Comment by SargeroDeV** [comment]
> Anyway, the idea is that it does not make several DELETE or several INSERT either, beyond the things that you mentioned. The idea is that it groups them, whenever possible, to reduce the amount of queries, inside the emulator.

EN: Of course, that was just one example.
ES: claro, solo es un ejemplo.

**Comment by pangolp** [comment]
Due to the time that has elapsed, and the requested changes have not been made, the pull request is closed. Then, in the future, if the changes are made, it can be reviewed again. Thanks anyway for the contribution, and we hope that in the future, you will take into consideration the suggestions made when creating a pull request, so that we all have the same way of working.

**Pull request closed without merging**